### ISSUE TITLE
Mac OS X fails to build with SDL or QT installed from homebrew

### ISSUE THREAD

**Issue by mikeage** [chasonr/nethack-3.6.0-patches#1]
Thanks for an awesome job on this porting! 

Building on Mac OS X (El Capitan, but I don't believe it matters), I noticed that the macosx10.10 hint file doesn't support SDL2 or QT that was installed via homebrew. Nor, for QT, does it support setting a custom QTDIR (this came from commit 6062fb1c).

For SDL, I just went ahead and installed the official SDL frameworks in /Library/Frameworks, so I'm not sure what exactly needs to be done; for QT, I removed the hardcoded setting QTDIR from the hints file, and changed:

QT_FRAMEWORKS_DIR = $(QTDIR)/Frameworks # /Library/Frameworks

I'm not sure what the best way to support either official installations or homebrew is (I'm pretty good with make on Linux systems, but I'm still figuring out how exactly Mac OS puts everything together), so I'm not submitting a patch. Just pointing out something that might be helpful!

Again, from someone whose imagination doesn't quite stretch far enough to visualize a @ or a ( or ", thanks for letting me enjoy Nethack 3.6!

**Comment by chasonr** [comment]
Thank you for noting what commit you used; I've done lots more since that one, though I don't think any of them address the issues you've noted.

My Mac is still running Yosemite and has SDL2, SDL2_image and various Qt 4 packages under /Library/Frameworks. By "homebrew", perhaps you mean that you built one or more of these packages from source?

Could you attach the hints file that finally worked for you?

**Comment by mikeage** [comment]
Just to be clear; the commit I was using was e6c14f4. Commit 6062fb1 was where you added a hardcoded QTDIR. 

Homebrew is http://brew.sh ; it's a pretty popular system for installing packages on Mac OS (similar to MacPorts, if you've ever used that). The biggest change for this context is  that everything is installed under /usr/local; for example, I have 

lrwxr-xr-x 1 mikemi admin 48 Dec 30 13:41 /usr/local//Frameworks/QtCore.framework -> ../Cellar/qt/4.8.7_2/Frameworks/QtCore.framework

for QtCore.framework. 

The hints file I used for using QT from homebrew (like I said; for SDL2, I just downloaded the two .framework files (SDL2 and SDL2_image) and copied them to /Library/Frameworks is attached (renamed because github won't let me upload a file with that "extension")
[macosx10.10.txt](https://github.com/chasonr/nethack-3.6.0-patches/files/76480/macosx10.10.txt)
